auth/src/main/java/com/firebase/ui/auth/ui/accountlink/WelcomeBackPasswordPrompt.java

```diff
@@ -103,10 +103,10 @@ public void onClick(View view) {
             next(mEmail, mPasswordField.getText().toString());
         } else if (id == R.id.trouble_signing_in) {
             startActivity(RecoverPasswordActivity.createIntent(
-                    getApplicationContext(),
+                    this,
                     mActivityHelper.getFlowParams(),
                     mEmail));
-            finish(ResultCodes.OK, new Intent());
+            finish(ResultCodes.CANCELED, IdpResponse.getErrorCodeIntent(ErrorCodes.UNKNOWN_ERROR));
```

Comment:
I believe this was also a mistake, the user is not signed in so we shouldn't finish with OK.